app/javascript/utilities/__tests__/textAreaUtils.test.js

```diff
@@ -62,16 +67,124 @@ describe('getMentionWordData', () => {
   });
 });
 
-describe('getIndexOfLineStart', () => {
-  it('returns 0 for empty text', () => {
-    expect(getIndexOfLineStart('', 0)).toEqual(0);
+describe('getLastIndexOfCharacter', () => {
+  it('returns -1 when content is empty', () => {
+    expect(
+      getLastIndexOfCharacter({
+        content: '',
+        selectionIndex: 0,
+        character: 'f',
+      }),
+    ).toEqual(-1);
   });
 
-  it('returns start index of 0 for a single line of text', () => {
-    expect(getIndexOfLineStart('something', 5)).toEqual(0);
+  it("returns -1 for a character that isn't present", () => {
+    expect(
+      getLastIndexOfCharacter({
+        content: 'abcde',
+        selectionIndex: 4,
+        character: 'f',
+      }),
+    ).toEqual(-1);
```

Comment:
Really appreciate the legibility of these tests.  I'm rather weak in JS, but this is quite readable for me.